### ISSUE TITLE
xGESVDX binding for faster low-rank SVD

### ISSUE THREAD

**Issue by mrader1248** [scipy/scipy#8231]
It would be nice to have a binding for the LAPACK functions xGESVDX to perform SVDs for only a part of the spectrum. A parameter `svdvals` could be introduced for the function `scipy.linalg.svd` (like there is the parameter `eigvals` for `scipy.linalg.eigh`) to select the singular values.

If it helps, I have implemented a binding to xGESVDX for my own needs:
[https://github.com/mrader1248/pylapack](https://github.com/mrader1248/pylapack)

**Comment by ilayn** [comment]
Since the number of LAPACK routines are too many, it is not so easy to prioritize what to add, hence LAPACK bindings are added in a kind-of *"we need this"* basis.  If you can submit a PR for this that would be great. 

All the LAPACK f2py signatures are located in `flapack.pyf.src` and you can add the signatures to that. As an example, the [`GESVD` is here ](https://github.com/scipy/scipy/blob/master/scipy/linalg/flapack.pyf.src#L1351). 

f2py syntax is quite cryptic so if you need to familiarize yourself with it please let us know. I've been there :)

**Comment by mrader1248** [comment]
So I would have to add the xGESVDX signatures to `flapack.pyf.src` and then I could use the function `get_lapack_funcs` to call xGESVDX from Python?

**Comment by ilayn** [comment]
Yes. Or if you are sure about the data type, you can directly call `scipy.linalg.lapack.zgesvdx(....)` or any other `<s/d/c>` flavor.
